vignettes/package.Rmd

```diff
@@ -2,119 +2,250 @@
 title: "Using reticulate in an R Package"
 output:
   rmarkdown::html_vignette
+editor_options:
+  markdown:
+    wrap: 80
 vignette: >
   %\VignetteIndexEntry{Using reticulate in an R Package}
   %\VignetteEngine{knitr::rmarkdown}
   %\VignetteEncoding{UTF-8}
 ---
 
+## Declaring Python Requirements
 
-## Delay Loading Python Modules
+Beginning with Reticulate version 1.41, R packages can declare Python
+requirements with `py_require()`. Reticulate can then resolve these requirements
+at runtime using an ephemeral Python environment.
 
-If you write an R package that wraps one or more Python packages, it's likely that you'll be importing Python modules within the `.onLoad` method of your package so that you can have convenient access to them within the rest of the package source code.
+Calling `py_require()` is always recommended because it has no effect if the
+user is using a manually managed Python environment, making it generally safe to
+use.
 
-When you do this, you should use the `delay_load` flag to the `import()` function, for example:
+### Typical Usage
 
-``` r
-# global reference to scipy (will be initialized in .onLoad)
-scipy <- NULL
+`py_require()` is typically called from `.onLoad()`, like this:
 
+``` r
 .onLoad <- function(libname, pkgname) {
-  # use superassignment to update global reference to scipy
-  scipy <<- reticulate::import("scipy", delay_load = TRUE)
+  reticulate::py_require("scipy")
 }
 ```
 
-Using the `delay_load` flag has two important benefits:
+It can also be called from other package functions. For example, the `keras3` R
+package defines a `use_backend()` function that calls `py_require()` with
+different values based on the user-requested backend:
```

Comment:
Can we emphasize that the intent here that:

- The `tensorflow` package provides a default backend, which `keras` makes available by default,
- Other packages (e.g. `jax`) provide alternate backends, which can be selected if desired by the user.

Reply:
I think this is also pointed out in the comments but would be good to emphasize here.